inception-recommendation/src/main/java/de/tudarmstadt/ukp/inception/recommendation/project/EvaluationSimulationPanel.java

```diff
@@ -0,0 +1,300 @@
+    {
+        super(aId);
+        project = aProject;
+
+        Form<Recommender> form = new Form<>(FORM);
+        add(form);
+
+        chartContainer = new Label(CHART_CONTAINER);
+        chartContainer.setOutputMarkupId(true);
+        form.add(chartContainer);
+
+        form.add(new AjaxButton(SIMULATION_START_BUTTON)
+        {
+            private static final long serialVersionUID = -3902555252753037183L;
+
+            @Override
+            protected void onError(AjaxRequestTarget aTarget)
+            {
+                aTarget.addChildren(getPage(), IFeedback.class);
+            }
+
+            @Override
+            protected void onAfterSubmit(AjaxRequestTarget aTarget)
+            {
+                //get the selected recommender
+                IModel<Recommender> recommenderModel = aRecommenderEditorPanel
+                        .getRecommenderModel();
+                
+                //there must be some recommender selected by the user on the UI
+                if (recommenderModel.getObject() == null
+                        || recommenderModel.getObject().getTool() == null) {
+                    log.error("Please select a recommender from the list");
+                    error("Please select a recommender from the list");
+                    aTarget.addChildren(getPage(), IFeedback.class);
+                    return;
+                }
+
+                Map<SourceDocument, AnnotationDocument> listAllDocuments = documentService
+                        .listAllDocuments(project, userDao.getCurrentUser());
+
+                //create a list of CAS from the pre-annotated documents of the project
+                List<CAS> casList = new ArrayList<>();
+                listAllDocuments.forEach((source, annotation) -> {
+                    try {
+                        CAS cas = documentService.createOrReadInitialCas(source).getCas();
+                        casList.add(cas);
+                    }
+                    catch (IOException e1) {
+                        log.error("Unable to render chart", e1);
+                        error("Unable to render chart: " + e1.getMessage());
+                        aTarget.addChildren(getPage(), IFeedback.class);
+                        return;
+                    }
+                });
+                
+                IncrementalSplitter splitStrategy = new IncrementalSplitter(TRAIN_PERCENTAGE,
+                        INCREMENT, LOW_SAMPLE_THRESHOLD);
+
+                RecommendationEngine recommender = getRecommendationEngine(recommenderModel);
+                if (recommender == null)
+                {
+                    log.warn("Unknown Recommender selected");
+                    warn("Unknown Recommender selected");
+                    aTarget.addChildren(getPage(), IFeedback.class);
+                    return;
+                }
+                
+                StringBuilder dataColumns = new StringBuilder();
```

Comment:
Please re-use the `LearningCurveChartPanel` instead of duplicating the code.

Reply:
Resolved